src/utils/compileManager.ts

```diff
@@ -2,23 +2,133 @@ import * as vscode from 'vscode';
 import { RemoteFileSystemProvider, parseUri } from '../provider/remoteFileSystemProvider';
 import { ROOT_NAME, ELEGANT_NAME, OUTPUT_FOLDER_NAME } from '../consts';
 import { PdfDocument } from '../provider/pdfViewEditorProvider';
-import { SyncCodeResponseSchema, SyncPdfResponseSchema } from '../api/base';
+import { LatexParser, ErrorSchema } from './compileLogParser';
 
-export const pdfViewRecord:{
-    [key:string]: {
-        [key:string]: {doc:PdfDocument, webviewPanel:vscode.WebviewPanel}
+// map string level to severity
+const severityMap: Record<string, vscode.DiagnosticSeverity> = {
+    error: vscode.DiagnosticSeverity.Error,
+    warning: vscode.DiagnosticSeverity.Warning,
+    information: vscode.DiagnosticSeverity.Information,
+};
+
+
+export const pdfViewRecord: {
+    [key: string]: {
+        [key: string]: { doc: PdfDocument, webviewPanel: vscode.WebviewPanel }
     }
 } = {};
 
+
+
+class CompileDiagnosticProvider {
+    private diagnosticCollection = vscode.languages.createDiagnosticCollection(ROOT_NAME);
+    constructor(private readonly vfsm: RemoteFileSystemProvider) { };
+
+    private async readOutput(_uri: vscode.Uri, vfs: any) {
+        try {
```

Comment:
error handling is no needed here.

If `output.log` is absent, then there is no diagnostics to update.